core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -435,6 +462,8 @@ class KafkaApis(val requestChannel: RequestChannel,
       }
 
       def fetchResponseCallback(delayTimeMs: Int) {
+          trace(s"Sending fetch response to ${fetchRequest.clientId} with ${convertedResponseStatus.values.map(_.messages.size).sum}" +
```

Comment:
It's probably better to log the # of bytes in the messageSet instead of # of messages. The later will invoke the iterator and is more expensive.